js/main.js

```diff
@@ -0,0 +1,314 @@
+const buttons = document.getElementsByClassName("button");
+const digitButtons = document.getElementsByClassName("digitButton");
+const screen = document.getElementById("screen");
+const unblockedKeys = [8, 13, 16, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 114, 187];
+
+let calcType = null;
+let result = null;
+let factor2= null;
+let newNumber = true;
+let dropTheResult = false;
+
+const calculation = function () {
+    switch (calcType) {
+        case 1:
+            result += factor2;
+            break;
+        case 2:
+            result -= factor2;
+            break;
+        case 3:
+            result *= factor2;
+            break;
+        case 4:
+            result /= factor2;
+            break;
+        case 5:
+            result %= factor2;
+            break;
+        default:
+            screen.value = "ERROR";   
+        }
+}
+
+const noNoughtStartsInteger = function () {
+    if (screen.value.charAt(0) === "0" && screen.value.charAt(1) !== "." ) {
+        newNumber = true;
+    }
+}
+
+const digit = function (symbol) {
+    if (dropTheResult) {
+        result = null;
+        dropTheResult = false;
+    }
+
+    if (newNumber) {
+        screen.value = symbol;
+        newNumber = false;
+    } else {
+        screen.value += symbol;
+        newNumber = false;
+    }
+}
+
+const calculationType = function (typeNo) {
+        if (dropTheResult) {
+        dropTheResult = false;
+    }
+
+    if (result === null) {
+        result = parseFloat(screen.value);
+    }
+    calcType = typeNo;
+    newNumber= true;
+    factor2= null;
+}
+
+document.onkeydown = function (e) {                 //to block unnecessary keys on the keyboard
+    if (!unblockedKeys.includes(e.keyCode)) {
+          return false;
+    }    
+}
+
+buttons[0].addEventListener("click", function(e){      //c
+    screen.value = "";
+    dropTheResult = false;
+    factor2= null;
+    calcType = null;
+    result = null;
+    newNumber = true;
+    e.preventDefault();
+})
+
+buttons[1].addEventListener("click", function(e){      //<-
+    e.preventDefault();
+    if (!newNumber){
+        screen.value = screen.value.slice(0,-1);
+        if (screen.value === "") {
+            screen.value = "0"
+        } 
+    }    
+})
+
+buttons[2].addEventListener("click", function(e){      //%
+    e.preventDefault();
+    if (screen.value === "Infinity" || screen.value === "NaN" || screen.value === "undefined") {         
+        return false     
+    };
+    calculationType(5)
+})
+
```

Comment:
you could add one event listener for /, *, -, +, and % buttons. Look at example below that handle all numbers.